### ISSUE TITLE
Make the `@yamada-ui/infinite-scroll-area` component headless

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#1873]
script: make-component-headless
package: @yamada-ui/infinite-scroll-area

### Description
Currently, the `@yamada-ui/infinite-scroll-area` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/infinite-scroll-area` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/infinite-scroll-area` component exists in `packages/components/infinite-scroll-area/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

**Comment by Hayato1031** [comment]
Hi,
Can you asssign me to this issue?

**Comment by hirotomoyamada** [comment]
@Hayato1031 

Hi, thanks for the comment!

Yes, pls.

**Comment by Hayato1031** [comment]
I wanna ask you one.

In `packages/theme/src/components`, there are no file which is named `infinite-scroll-area.ts`.
Do I have to make the file?

**Comment by Hayato1031** [comment]
@hirotomoyamada 
sorry, I forfot mention to you.
Tell me what do I have to do to this component.

**Comment by hirotomoyamada** [comment]
@Hayato1031 

Hi, thanks for the comment!

Sorry. The `InfiniteScrollArea` component isn't themed, right? First of all, we need to theme it, but it doesn't seem like it's at the level of a "good first issue".

**Comment by hirotomoyamada** [comment]
Related #2030

**Comment by Hayato1031** [comment]
@hirotomoyamada 

Thanks for your reply.
I got it, I stop working on this issue until you solve the problem.
If there is something what I can do, please tell me.

**Comment by hirotomoyamada** [comment]
@Hayato1031 

#2031

Once this PR is merged, you can work on it.

**Comment by Hayato1031** [comment]
@hirotomoyamada 
I cant believe that how fast you work on it.
Thanks for your cooperation.

**Comment by Hayato1031** [comment]
@hirotomoyamada 
Hi
Thanks for working on it.

I have a question.
You add theming to the `infinite-scroll-area` component on the PR, but the change is merged to the `main` branch right?

For working on this issue, I have to make new branch from `v2.0`.
But there are no changes on `v2.0` branch, therefore still I cant make new working branch.
What can I do for it?

**Comment by hirotomoyamada** [comment]
@Hayato1031 

Hi, thanks for the comment!

I have now merged main:)

**Comment by Hayato1031** [comment]
@hirotomoyamada 

You mean, I should make new branch from `main`, and make PR to `V2.0`?

I thought that I have to make new branch from `V2.0`, and make PR to `V2.0`.

**Comment by hirotomoyamada** [comment]
@Hayato1031 

It has been merged into v2.0, so please create a branch from v2.0.

**Comment by hirotomoyamada** [comment]
Fixed #2058
